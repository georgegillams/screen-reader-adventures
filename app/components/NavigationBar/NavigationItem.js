import React from 'react';
import PropTypes from 'prop-types';

import GGButton from 'components/GGButton';

const NavigationItem = props => {
  const { name, linkUrl, className, ...rest } = props;

  return (
    <GGButton href={linkUrl} bouncy buttonClassName={className} {...rest}>
      {name}
    </GGButton>
  );
};

NavigationItem.propTypes = {
  name: PropTypes.string,
  linkUrl: PropTypes.string,
  className: PropTypes.string,
};

NavigationItem.defaultProps = {
  name: null,
  linkUrl: null,
  className: null,
};

export default NavigationItem;
