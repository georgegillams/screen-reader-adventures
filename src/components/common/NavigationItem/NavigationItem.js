import React from 'react';
import PropTypes from 'prop-types';
import Button from 'components/common/Button';

const NavigationItem = props => {
  const { name, href, ...rest } = props;

  return (
    <Button href={href} bouncy {...rest}>
      {name}
    </Button>
  );
};

NavigationItem.propTypes = {
  name: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
};
NavigationItem.defaultProps = {};

export default NavigationItem;
