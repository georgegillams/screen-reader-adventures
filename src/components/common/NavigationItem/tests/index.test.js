import React from 'react';

import { render } from '@testing-library/react';

import NavigationItem from '..';

describe('<NavigationItem />', () => {
  it('Should render correctly', () => {
    const { container } = render(<NavigationItem name="name" href="/test"></NavigationItem>);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly with custom className', () => {
    const { container } = render(
      <NavigationItem name="name" href="/test" className={'test-custom-classname'}></NavigationItem>
    );

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly hrefExternal', () => {
    const { container } = render(<NavigationItem name="name" href="/test" hrefExternal></NavigationItem>);

    expect(container).toMatchSnapshot();
  });
});
