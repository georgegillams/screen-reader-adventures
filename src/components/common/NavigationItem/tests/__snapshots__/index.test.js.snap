// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItem /> Should render correctly 1`] = `
<div>
  <a
    aria-label="name"
    class="button__a button__outer button__outer--dark-text button__outer--bouncy button__outer--link"
    href="/test"
  >
    <span
      class="button__child"
    >
      name
    </span>
  </a>
</div>
`;

exports[`<NavigationItem /> Should render correctly hrefExternal 1`] = `
<div>
  <a
    aria-label="name"
    class="button__a button__outer button__outer--dark-text button__outer--bouncy button__outer--link"
    href="/test"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      class="button__child"
    >
      name
    </span>
  </a>
</div>
`;

exports[`<NavigationItem /> Should render correctly with custom className 1`] = `
<div>
  <a
    aria-label="name"
    class="button__a button__outer button__outer--dark-text button__outer--bouncy button__outer--link test-custom-classname"
    href="/test"
  >
    <span
      class="button__child"
    >
      name
    </span>
  </a>
</div>
`;
